File: 287-find-the-duplicate-number/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        unordered_map<int,int>hash;
        int n=nums.size();
        for(int i=0;i<n;i++){
          hash[nums[i]]++;  
        }
        for(auto i:hash){
            if(i.second>1){
                return i.first;
            }
        }
        return -1;
    }
};

File: 287-find-the-duplicate-number/find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        Set<Integer> st= new HashSet<>();
        int i=0;
        while(i<nums.length){
            if(st.contains(nums[i])) return nums[i];
            st.add(nums[i]);
            i++;

        }
        return -1;
    }
}